### PR TITLE
Update Python 3.11.0 tgz checksum

### PR DESCRIPTION
**Pull request by playpauseandstop**
Make sure you have checked all steps below.

### Prerequisite
* [ ] Please consider implementing the feature as a hook script or plugin as a first step.
  * pyenv has some powerful support for plugins and hook scripts. Please refer to [Authoring plugins](https://github.com/pyenv/pyenv/wiki/Authoring-plugins) for details and try to implement it as a plugin if possible.
* [ ] Please consider contributing the patch upstream to [rbenv](https://github.com/rbenv/rbenv), since we have borrowed most of the code from that project.
  * We occasionally import the changes from rbenv. In general, you can expect changes made in rbenv will be imported to pyenv too, eventually.
  * Generally speaking, we prefer not to make changes in the core in order to keep compatibility with rbenv.
* [ ] My PR addresses the following pyenv issue (if any)
  - Closes https://github.com/pyenv/pyenv/issues/XXXX

### Description

**TLDR:** For some reason checksum for `Python-3.11.0.tgz` file introduced in 0726e02e3a1b638f6442dffd880abf256e9c2bf7 is invalid. Updated that checksum to ensure `pyenv install 3.11.0` works well on systems without tar.xz support.

---

I have tried to run `pyenv install 3.11.0` on one of my Ubuntu 18.04 boxes and it failed with,

```
BUILD FAILED (Ubuntu 18.04 using python-build 20180424)

Inspect or clean up the working tree at /tmp/python-build.20221025231701.62
Results logged to /tmp/python-build.20221025231701.62.log

Last 10 log lines:
tmp/python-build.20221025231701.62 ~

checksum mismatch: Python-3.11.0.tar.gz (file is corrupt)
expected 20d77729a64b2a300f08717f7ffcb8da189d02e7c51e6e4a06c0340b619cbf32, got 64424e96e2457abbac899b90f9530985b51eef2905951febd935f0e73414caeb
```

This makes me think that, for some reason, checksum for `Python-3.11.0.tgz` file has been invalid in pyenv repo. I can approve this by running following command on same Ubuntu 18.04 box,

```
$ curl -sSL https://www.python.org/ftp/python/3.11.0/Python-3.11.0.tgz | sha256sum -
64424e96e2457abbac899b90f9530985b51eef2905951febd935f0e73414caeb  -
```

And following on my Mac M1,

```
❯ curl -sSL https://www.python.org/ftp/python/3.11.0/Python-3.11.0.tgz | shasum -a 256 -
64424e96e2457abbac899b90f9530985b51eef2905951febd935f0e73414caeb  -
```

Cause of that, I have decided to update `3.11.0` build file with updated checksum for `Python-3.11.0.tgz` file.

ps. And yes, installing `3.11.0` via `tar.xz` file works well for me on my Mac M1 and other Ubuntu boxes with XZ (`xz-utils`) installed.

### Tests
- [ ] My PR adds the following unit tests (if any)

### CODE REVIEW

**Review by humitos — APPROVED** [review]
I'm facing the same issue and this PR solves the problem 💯

**Review by exilvm — APPROVED** [review]

**Comment by humitos** [comment]
Hrm... Interesting. I installed 3.11.0 on CircleCI using pyenv _without this patch_ and it did work! Which is confusing 🤔

**Comment by exilvm** [comment]
Confusing indeed.
For me it does work inside a Codespaces/Devcontainer environment (which is based on Docker), but in Docker it throws checksum error.

**Comment by playpauseandstop** [comment]
Hi @humitos & @exilvm 

Thanks for your messages. 

The error happens only if your system does not have `xz-utils` installed and as result next if statement is falsy: https://github.com/pyenv/pyenv/blob/0726e02e3a1b638f6442dffd880abf256e9c2bf7/plugins/python-build/share/python-build/3.11.0#L5

In that case it tries to download `Python-3.11.0.tgz` and afterwards fail with `checksum mismatch`

<details>
<summary>Dockerfile and reproduction steps</summary>

**`Dockerfile.pyenv-3.11.0-tgz`**

```Dockerfile
FROM ubuntu:18.04

RUN apt-get update -yqq \
    && apt-get upgrade -yq \
    && apt-get install -yq --no-install-recommends ca-certificates curl git

ENV PYENV_ROOT="${HOME}/.pyenv"
RUN curl https://pyenv.run | bash

ENV PATH="${PYENV_ROOT}/bin:${PATH}"
RUN pyenv install 3.11.0
```

```
❯ docker build -f Dockerfile.pyenv-3.11.0-tgz -t pyenv-3.11.0-tgz:latest .
[+] Building 80.7s (7/7) FINISHED                                                                                                                                                                                                             
 => [internal] load build definition from Dockerfile.pyenv-3.11.0-tgz                                                                                                                                                                    0.1s
 => => transferring dockerfile: 341B                                                                                                                                                                                                     0.1s
 => [internal] load .dockerignore                                                                                                                                                                                                        0.0s
 => => transferring context: 2B                                                                                                                                                                                                          0.0s
 => [internal] load metadata for docker.io/library/ubuntu:18.04                                                                                                                                                                          0.0s
 => CACHED [1/4] FROM docker.io/library/ubuntu:18.04                                                                                                                                                                                     0.0s
 => [2/4] RUN apt-get update -yqq     && apt-get upgrade -yq     && apt-get install -yq --no-install-recommends ca-certificates curl git                                                                                                60.9s
 => [3/4] RUN curl https://pyenv.run | bash                                                                                                                                                                                             13.3s 
 => ERROR [4/4] RUN pyenv install 3.11.0                                                                                                                                                                                                 6.1s 
------                                                                                                                                                                                                                                        
 > [4/4] RUN pyenv install 3.11.0:                                                                                                                                                                                                            
#6 0.560 Downloading Python-3.11.0.tar.gz...                                                                                                                                                                                                  
#6 0.561 -> https://www.python.org/ftp/python/3.11.0/Python-3.11.0.tgz                                                                                                                                                                        
#6 5.991                                                                                                                                                                                                                                      
#6 5.991 BUILD FAILED (Ubuntu 18.04 using python-build 20180424)                                                                                                                                                                              
#6 6.005 
#6 6.008 Inspect or clean up the working tree at /tmp/python-build.20221026172116.62
#6 6.011 Results logged to /tmp/python-build.20221026172116.62.log
#6 6.011 
#6 6.011 Last 10 log lines:
#6 6.014 /tmp/python-build.20221026172116.62 /
#6 6.014 
#6 6.014 checksum mismatch: Python-3.11.0.tar.gz (file is corrupt)
#6 6.014 expected 20d77729a64b2a300f08717f7ffcb8da189d02e7c51e6e4a06c0340b619cbf32, got 64424e96e2457abbac899b90f9530985b51eef2905951febd935f0e73414caeb
#6 6.014 
------
executor failed running [/bin/sh -c pyenv install 3.11.0]: exit code: 1
```

---

</details>

But if your system has `xz-utils` (or other package), which brings `.tar.xz` support - all works well, as it now attempts to download `Python-3.11.0.tar.xz` file with proper checksum: https://github.com/pyenv/pyenv/blob/0726e02e3a1b638f6442dffd880abf256e9c2bf7/plugins/python-build/share/python-build/3.11.0#L6

<details>
<summary>Dockerfile & reproduction steps</summary>

**`Dockerfile.pyenv-3.11.0-tar-xz`**

```Dockerfile
FROM ubuntu:18.04

RUN apt-get update -yqq \
    && apt-get upgrade -yq \
    && apt-get install -yq --no-install-recommends ca-certificates curl git xz-utils

ENV PYENV_ROOT="${HOME}/.pyenv"
RUN curl https://pyenv.run | bash

ENV PATH="${PYENV_ROOT}/bin:${PATH}"
RUN pyenv install 3.11.0
```

**Diff between Dockerfiles**:

```diff
5c5
<     && apt-get install -yq --no-install-recommends ca-certificates curl git
---
>     && apt-get install -yq --no-install-recommends ca-certificates curl git xz-utils
```

```
❯ docker build -f Dockerfile.pyenv-3.11.0-tar-xz -t pyenv-3.11.0-tar-xz:latest .
[+] Building 76.7s (7/7) FINISHED                                                                                                                                                                                                             
 => [internal] load build definition from Dockerfile.pyenv-3.11.0-tar-xz                                                                                                                                                                 0.1s
 => => transferring dockerfile: 353B                                                                                                                                                                                                     0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                        0.0s
 => => transferring context: 2B                                                                                                                                                                                                          0.0s
 => [internal] load metadata for docker.io/library/ubuntu:18.04                                                                                                                                                                          0.0s
 => CACHED [1/4] FROM docker.io/library/ubuntu:18.04                                                                                                                                                                                     0.0s
 => [2/4] RUN apt-get update -yqq     && apt-get upgrade -yq     && apt-get install -yq --no-install-recommends ca-certificates curl git xz-utils                                                                                       56.3s
 => [3/4] RUN curl https://pyenv.run | bash                                                                                                                                                                                             14.0s 
 => ERROR [4/4] RUN pyenv install 3.11.0                                                                                                                                                                                                 6.3s 
------                                                                                                                                                                                                                                        
 > [4/4] RUN pyenv install 3.11.0:                                                                                                                                                                                                            
#6 0.543 Downloading Python-3.11.0.tar.xz...                                                                                                                                                                                                  
#6 0.544 -> https://www.python.org/ftp/python/3.11.0/Python-3.11.0.tar.xz                                                                                                                                                                     
#6 5.488 Installing Python-3.11.0...               
...
```

_Build is also failed as additional build dependencies does not installed_

---

</details>

To sum up, if you need to have 3.11.0 be installed via pyenv - make sure `xz-utils` or similar package has been installed at your system.

Otherwise, please wait before this PR will be reviewed and merged 🤝

**Comment by humitos** [comment]
@playpauseandstop excellent explanation! Thanks a lot.
